### PR TITLE
HF shower library interface vers03

### PR DESCRIPTION
**Pull request by lveldere**
I open this pr to enable a discussion.
A proper description will follow later.

### CODE REVIEW

File: SimG4CMS/Calo/src/HFShowerLibrary.cc [hunk 1]
```diff
@@ -211,6 +211,21 @@ std::vector<HFShowerLibrary::Hit> HFShowerLibrary::getHits(G4Step * aStep,
...
+                           << hitPoint.x() << "," << hitPoint.y() << ","
+                           << hitPoint.z() << " (" << zoff
+                           << ")   sphi,cphi,stheta,ctheta  = " << sin(momDir.phi())
+                           << ","  << cos(momDir.phi()) << ", " << sin(momDir.theta()) 
+                           << "," << cos(momDir.theta());
+#endif
+
   std::vector<HFShowerLibrary::Hit> hit;
   ok = false;
   if (parCode == pi0PDG || parCode == etaPDG || parCode == nuePDG ||
```
Comment (lveldere) [inline SimG4CMS/Calo/src/HFShowerLibrary.cc]: @Vlandr57 

move the parCode condition to the fillHits function,
this allows you to remove the parCode condition from
FastHFShowerLibrary

File: FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h [hunk 2]
```diff
@@ -45,61 +46,32 @@ class FastHFShowerLibrary {
...
 public:
 
-  struct Hit {
-    Hit() {}
-    G4ThreeVector             position;
-    int                       depth;
-    double                    time;
-  };
-
   void                initRun();
```
Comment (lveldere) [inline FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h]: @Vlandr57 

If you move the parCode condition into HFShowerLibrary::fillHits,
FastHFShowerLibrary::initRun can be removed.

File: FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h [hunk 3]
```diff
@@ -45,61 +46,32 @@ class FastHFShowerLibrary {
...
   void       const    initHFShowerLibrary(const edm::EventSetup& );
   void                recoHFShowerLibrary(const FSimTrack &myTrack);
-  std::vector<Hit>    getHits(const G4ThreeVector & p, const G4ThreeVector & v,
-                              int parCode, double parEnergy, bool &ok, 
-                              double weight, bool onlyLong=false);
-  const std::map<CaloHitID,float>& getHitsMap() { return hitMap; };
-  bool                isItinFidVolume (G4ThreeVector&);
 
-protected:
+  std::vector<HFShowerLibrary::Hit> getHits(G4ThreeVector & p, G4ThreeVector & v,
```
Comment (lveldere) [inline FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h]: @Vlandr57 

when you move the parCode condition into HFShowerLibrary::fillHits,
the parcode condition can be removed from FastHFShowerLibrary::getHits.

The 2 remaining lines of code inside this function better move to FastHFShowerLibrary::recoHFShowerLibrary

File: FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h [hunk 4]
```diff
@@ -45,61 +46,32 @@ class FastHFShowerLibrary {
...
-  void                loadEventInfo(TBranch *);
-  void                interpolate(int, double);
-  void                extrapolate(int, double);
-  void                storePhoton(int j);
-  std::vector<double> getDDDArray(const std::string&, const DDsvalues_type&,
-				  int&);
+  const std::map<CaloHitID,float>& getHitsMap() { return hitMap; };
 
-private:
+  bool                isItinFidVolume (G4ThreeVector&);
```
Comment (lveldere) [inline FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h]: @Vlandr57 

remove this function and replace it inside FastHFShowerLibrary::recoHFShowerLibrary
with 
if( !applyFidCut || applyFidCut(pos) )

**Comment by civanch** [comment]
@lveldere , @Vlandr57 , in FullSim the default SL file is for run-1 (in this PR - for run-2); in general, this PR seems to be OK for FullSim.

**Comment by lveldere** [comment]
Hi Vladimir

Vladimir Andreev has implemented such that for run1 we use GFLASH, as
before,
while for run2 the run2 SL is used. This is fine for now.
It might be that we get rid of GFLASH for run1 at a later stage, then we
have to keep this in mind.

Lukas

File: FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h (hunk 4 above)
Comment (Vlandr57) [inline FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h]: Yes, it is acceptable

File: FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h (hunk 3 above)
Comment (Vlandr57) [inline FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h]: Yes. it is possible to remove FastHFShowerLibrary::getHits function

File: SimG4CMS/Calo/src/HFShowerLibrary.cc (hunk 1 above)
Comment (Vlandr57) [inline SimG4CMS/Calo/src/HFShowerLibrary.cc]: The code for particles in FullSim is initiated with initRun(G4ParticleTable * theParticleTable) function,
so for FastSim need to write some interface for this case.

Comment (lveldere) [inline SimG4CMS/Calo/src/HFShowerLibrary.cc]: right.

@civanch, can we give the *PDG variabels proper values inside the constructor of HFShowerLibrary?

Comment (lveldere) [inline SimG4CMS/Calo/src/HFShowerLibrary.cc]: @civanch , is there anything against initialising the *PDG variables in the constructor to the usual PDG codes?
@Vlandr57 , could you take care of that?

Comment (civanch) [inline SimG4CMS/Calo/src/HFShowerLibrary.cc]: @lveldere , @Vlandr57, of course it is correct to move these lines to the constructor. If later we will need to use pdg codes from Geant4, then I would say Geant4 particle types are static, so if one access whatever Geant4 particle type it is already defined.

**Comment by lveldere** [comment]
moves to #8591

**Pull request closed without merging**